### DISCUSSION TITLE
how to use useEffect()  for Fetching data ?

### DISCUSSION THREAD

**Discussion by devrosada** [iamrosada/friendly-venus#5] [Q&A] [answered]
i have a function  FetchEmployees(), and   i want to use him to call the `useEffect`

**Comment by iamrosada** [comment, marked as answer]
Fetching data
useEffect() can perform data fetching side-effect.

The following component FetchEmployees fetches the employees list over the network

` 
import { useEffect, useState } from 'react';

function FetchEmployees() {
  const [employees, setEmployees] = useState([]);

  useEffect(() => {
    async function fetchEmployees() {
      const response = await fetch('/employees');
      const fetchedEmployees = await response.json();
      setEmployees(fetchedEmployees);
    }

    fetchEmployees();
  }, []);

  return (
    <div>
      {employees.map(name => <div>{name}</div>)}
    </div>
  );
}
`
